components/Main.js

```diff
@@ -66,12 +65,59 @@ export default () => {
   }, [request.name]);
 
   // Parse dependencies for the current code
+
+  const requestFileRef = react.useRef(null);
+  const worker = react.useRef(null);
+
+  // set up webworker and post url
+  react.useEffect(() => {
+    if (request.file && requestFileRef.current !== request.file) {
+      if (worker.current) {
+        worker.current.terminate();
+      }
+      requestFileRef.current = request.file;
+      worker.current = new Worker('./utils/recursiveDependencyFetchWorker.js');
+      worker.current.postMessage('https://unpkg.com/' + request.path);
+    }
+  }, [request.path, request.file, worker.current]);
+
+  // Set up listener for messages from the webworker
   react.useEffect(() => {
-    if (request.file)
-      parseDependencies('https://unpkg.com/' + request.path).then(cache =>
-        dispatch({ type: 'setCache', payload: cache })
-      );
-  }, [request.name, request.version, request.file]);
+    let buffer = {};
+    if (worker.current) {
+      worker.current.addEventListener('message', e => {
+        if (e.data.url === 'https://unpkg.com/' + request.path) {
+          dispatch({
+            type: 'setCache',
+            payload: { [e.data.url]: e.data },
+          });
+        } else {
+          buffer = { ...buffer, [e.data.url]: e.data };
+        }
+      });
+      setInterval(() => {
+        if (Object.keys(buffer).length > 0) {
+          dispatch({
+            type: 'setCache',
+            payload: buffer,
+          });
+          buffer = {};
+        }
+      }, 1000);
+    }
+    return () =>
+      worker.current &&
+      worker.current.removeEventListener('message', e => {
+        if (e.data.url === 'https://unpkg.com/' + request.path) {
```

Comment:
Can we put this in a helper function so it's easily reference-able for the `add` and `remove` event listeners?